### PR TITLE
Add test for permanent redirects, plus some minor changes

### PR DESCRIPTION
**Pull request by timo0**
Hi!
Thanks for adding a pull request for Dolphin Platform :)
Please add a short description that explains the changes of your pull request.
We try to review your pull request as fast as possible.

---
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/904)

### CODE REVIEW

File: platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java [hunk 1]
```diff
@@ -17,16 +17,13 @@
 
 import com.canoo.dp.impl.platform.core.http.HttpHeaderConstants;
 import com.canoo.platform.client.PlatformClient;
-import com.canoo.platform.core.http.BadResponseException;
-import com.canoo.platform.core.http.ByteArrayProvider;
-import com.canoo.platform.core.http.ConnectionException;
-import com.canoo.platform.core.http.HttpClient;
-import com.canoo.platform.core.http.HttpResponse;
+import com.canoo.platform.core.http.*;
```
Comment (hendrikebbers) [inline platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java]: no star imports please

File: platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java [hunk 2]
```diff
@@ -42,6 +39,8 @@
 import static org.hamcrest.Matchers.notNullValue;
 
 public class HttpClientTests {
+    protected static final String STD_GET_RESPONSE = "Spark Server for HTTP client integration tests";
```
Comment (hendrikebbers) [inline platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java]: why not final?

File: platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java [hunk 3]
```diff
@@ -88,29 +88,6 @@ public void destroySpark() {
         Spark.stop();
     }
 
-    @Test
-    public void testSimpleGet() throws Exception {
```
Comment (hendrikebbers) [inline platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java]: mmh, does it really make sense to remove this test?

File: platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java (hunk 2 above)
Comment (timo0) [reply platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java]: this is final.. you mean private?

File: platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java (hunk 3 above)
Comment (timo0) [reply platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java]: This is the same as the test bellow only that you call .get twice.. was this your intention?

File: platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java (hunk 1 above)
Comment (timo0) [reply platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java]: done

File: platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java (hunk 3 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-client/src/test/java/com/canoo/dp/impl/platform/client/http/HttpClientTests.java]: ok, fine.

**Pull request merged**